src/hal/random.zig

```diff
@@ -0,0 +1,37 @@
+//! Random number generator (RNG) using the ROSC
+//!
+//! _WARNING_: This does not meet the requirements of randomness
+//! for security systems because it can be compromised, but it
+//! may be useful in less critical applications.
+
+const microzig = @import("microzig");
+const peripherals = microzig.chip.peripherals;
+
+/// Fill the buffer with up to buffer.len random bytes
+///
+/// rand uses the RANDOMBIT register of the ROSC as its source, i. e.,
+/// the system clocks _MUST_ run from the XOSC and/or PLLs.
+///
+/// _WARNING_: This function does not meet the requirements of randomness
+/// for security systems because it can be compromised, but it may be useful
+/// in less critical applications.
+pub fn rand(buffer: []u8) void {
```

Comment:
Add an assert to make sure the clocks are configured as you expect, and make this generator expose the stdlib's random interface so we can utilize a bunch of helper functions.